noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -198,75 +195,41 @@ pub contract NFT {
-        // and for this reason the protocol guarantees that nobody can front-run us in consuming the hiding point.
-        // This guarantee would break if `finalize_transfer_to_private` was not called in the same transaction. This
-        // however is not the flow we are currently concerned with. To support the multi-transaction flow we could
-        // introduce a `from` function argument, hash the x-coordinate with it and then repeat the hashing in
-        // `finalize_transfer_to_private`.
-        //
-        // We can also be sure that the `hiding_point_slot` will not overwrite any other value in the storage because
-        // in our state variables we derive slots using a different hash function from multi scalar multiplication
-        // (MSM).
-        let hiding_point_slot = note_setup_payload.hiding_point.x;
-
-        // We don't need to perform a check that the value overwritten by `_store_point_in_transient_storage_unsafe`
-        // is zero because the slot is the x-coordinate of the hiding point and hence we could only overwrite
-        // the value in the slot with the same value. This makes usage of the `unsafe` method safe.
-        NFT::at(context.this_address())
-            ._store_payload_in_transient_storage_unsafe(
-                hiding_point_slot,
-                note_setup_payload.hiding_point,
-                setup_log,
-            )
-            .enqueue(context);
-
-        hiding_point_slot
+    ) -> PartialNFTNote {
+        // We create a partial note with unpopulated/zero token id for 'to'
+        let partial_note = NFTNote::partial(
+            to,
+            storage.private_nfts.at(to).storage_slot,
+            context,
+            to,
+            context.msg_sender(),
+        );
+
+        NFT::at(context.this_address())._store_nft_set_partial_note(partial_note).enqueue(context);
+
+        partial_note
     }
     // docs:end:prepare_private_balance_increase
 
-    // TODO(#9375): Having to define the note log length here is very unfortunate as it's basically impossible for
-    // users to derive manually. This will however go away once we have a real transient storage since we will not need
-    // the public call and instead we would do something like `context.transient_storage_write(slot, payload)` and that
-    // will allow us to use generics and hence user will not need to define it explicitly. We cannot use generics here
-    // as it is an entrypoint function.
     // docs:start:store_payload_in_transient_storage_unsafe
     #[public]
     #[internal]
-    fn _store_payload_in_transient_storage_unsafe(
-        slot: Field,
-        point: Point,
-        setup_log: [Field; 9],
-    ) {
-        context.storage_write(slot, point);
-        context.storage_write(slot + aztec::protocol_types::point::POINT_LENGTH as Field, setup_log);
+    fn _store_nft_set_partial_note(partial_note: PartialNFTNote) {
```

Comment:
Do we still have some plan down the pipeline to somehow get rid of this public call?

Reply:
No, best thing we could do is have the library automatically inject it.

Reply:
Transient nullifier storage to avoid the public call, if the whole process is to be atomic in one tx 😏